File: projects/Strangers Calendar App/src/auth/phone_validation.js
|
// Phone number input and validation logic
const Joi = require('joi');

const phoneNumberSchema = Joi.string()
.pattern(/^\+1\d{10}$/)
.messages({
'string.pattern.base': 'Phone number must be in the format +1XXXXXXXXXX'
});

function validatePhoneNumber(phoneNumber) {
return phoneNumberSchema.validate(phoneNumber);
}

module.exports = { validatePhoneNumber };

File: backend/auth/phone_validation.py
|
# Phone number validation implementation
from flask import request, jsonify

def validate_phone_number(phone_number):
"""Validate phone number format."""
if not phone_number.isdigit():
return False
if len(phone_number) != 10:
return False
return True

@app.route('/validate-phone', methods=['POST'])
def validate_phone():
data = request.get_json()
phone_number = data.get('phone_number')

if validate_phone_number(phone_number):
return jsonify({'status': 'success'}), 200
else:
return jsonify({'status': 'error', 'message': 'Invalid phone number format'}), 400